include/z64.h

```diff
@@ -217,7 +217,7 @@ typedef struct {
     /* 0x4D */ char     unk_4D[0x03];
     /* 0x50 */ TargetContextEntry arr_50[3];
     /* 0x8C */ Actor*   unk_8C;
-    /* 0x90 */ Actor*   unk_90;
+    /* 0x90 */ Actor*   nearestEnemyBgm;
```

Comment:
Calling an actor pointer `nearestEnemyBgm` doesn't feel right to me, is it intended?

Reply:
Having both "nearest" and "bgm" in the name doesn't make much sense to me. I think it would make more sense as simply the nearest enemy actor (used for bgm but possibly other things as well). Same for sNearestEnemyBgmDistSq

Reply:
From my understanding, this does not track all/generic enemies as you specifically need both the `ACTOR_FLAG_0` (targetable) and `ACTOR_FLAG_2` (0x4 is documented as the enemy bgm flag) to count as an enemy that plays the enemy bgm as you approach. Not all enemies have the `ACTOR_FLAG_2` flag, so this is not pointing to a generic enemy, but rather a subset relating to bgm enemies.

That being said, there might be more to this as well. I just checked out @Dragorn421 `ACTOR_FLAG` PR to see if anything else is going on with `ACTOR_FLAG_2` that hasn't been reported yet. It does seem to do slightly more. In actor, the only other function it serves is to be required play the enemy target sfx instead of the "human" lock on sfx.

In player, it's also related to state flags `stateFlags1 & 0x10`, `stateFlags1 & 0x10000`, and `stateFlags2 & 0x2000`. So that's interesting to me.

Reply:
this is from the speedrunning communities notes so take with a grain of salt, but 0x10 is targeting an enemy, 0x10000 is targeting anything(?) and flags2 & 0x2000 is targeting something with switch targeting enabled specifically

Reply:
Nice, thanks for that. So while the details might not be exact, the common theme is targeting which makes a lot of sense since what I'm currently calling `nearestBgmEnemy` is stored in `TargetContext`.

Okay so with that, I think it's still worth making the distinction between "the nearest enemy to Link with the `ACTOR_FLAG_2` enemy bgm flag set" vs. simply "the nearest enemy to Link", as it might not always be the case that the closest enemy to Link has this flag set, meaning `nearestBgmEnemy` could be pointing to the second or third closest enemy to Link, all dependent on flags.

I guess 1) independent of the name, does this explanation of what's going on make sense? 2) Thoughts on the name itself?

Reply:
It makes sense in the most part, tho I'm not sure if we should trust the existing documentation on the enemy bgm flag part, but yeah if it's specifically for enemies with a bgm, then `nearestBgmEnemy` is probably fine

Reply:
That's fair to be cautious of the existing docs, but my current knowledge is entirely from the repo now. Setting the "nearestBgmEnemy", the target sfx, and the three stateFlags "possibly" related to targeting (haven't fully confirmed that myself, just the relation to the stateflags) seems to be the full usage of `ACTOR_FLAG_2`, assuming no instances were missed.

Other alternatives could be something like `closestBgmEnemy` or possibly just `bgmEnemy` if those sound nicer to you

Reply:
Does the audio system care about whether it's the nearest enemy? If not then `bgmEnemy` could be good enough

Reply:
I think a good compromise could be `bgmEnemy` with a brief comment explaining that its always going to be the closest one to you. 
The audio system doesnt seem to care, but it is a variable in the actor context so the implementation of it in the actor system is a little bit relevant in that regard imo

Reply:
The single parameter that the audio system cares about is "distance" to the nearest bgm enemy i.e.:
`Audio_SetNearestEnemyBgmVolume(sqrtf(globalCtx->actorCtx.targetCtx.nearestEnemyBgm->xyzDistToPlayerSq));`

This distance is used in setting both the volume of the "BGM_ENEMY" played on the sub-bgm player, but is also relevant to how the channels are split between the main-bgm player (for example, the dungeon music when no enemies are around) and the sub-bgm player.

I suppose it'd be fine just leaving it as `bgmEnemy` with the extra comment. There are a few places to remove the `Nearest` part,  I'll try to add comments where necessary